encode/vp8enc.c

```diff
@@ -0,0 +1,1074 @@
+
+}
+
+void vp8enc_destroy_buffers()
+{
+  VAStatus va_status;
+
+  for(int i = 0; i < vaapi_context.num_va_buffers; i++) {
+    if (vaapi_context.va_buffers[i] != VA_INVALID_ID) {
+      va_status = vaDestroyBuffer(vaapi_context.display, vaapi_context.va_buffers[i]);
+      CHECK_VASTATUS(va_status,"vaDestroyBuffer");
+      vaapi_context.va_buffers[i] = VA_INVALID_ID;
+    }
+  }
+
+  if (vaapi_context.codedbuf_buf_id != VA_INVALID_ID) {
+    va_status = vaDestroyBuffer(vaapi_context.display, vaapi_context.codedbuf_buf_id);
+    CHECK_VASTATUS(va_status,"vaDestroyBuffer");
+    vaapi_context.codedbuf_buf_id = VA_INVALID_ID;
+  }
+
+}
+void vp8enc_show_help ()
+{
+  printf("Usage: vp8enc_go <width> <height> <input_yuvfile> <output_vp8> additional_option\n");
+  printf("output_vp8 should use *.ivf\n");
+  printf("The additional option is listed\n");
+  printf("-f <frame rate> \n");
+  printf("--intra_period <key_frame interval>\n");
+  printf("--qp <quantization parameter> \n");
+  printf("--rcmode <rate control mode> 0: CQP, 1: CBR, 2: VBR\n");
+  printf("--fb <bitrate> (kbps unit)\n");
+  printf("--lf_level <loop filter level>  [0-63]\n");
+  printf("--hrd_win <num>  [1000-8000]\n");
+  printf("--vbr_max <num> (kbps unit. It should be greater than fb)\n");
+  printf("--fn_num <num>\n  how many frames to be encoded\n");
+  printf("--error_resilient Turn on Error resilient mode\n");
+  printf("--debug Turn debug info on\n");
+}
+
+void parameter_check(const char *param, int val, int min, int max)
+{
+  if(val < min || val > max)
+  {
+    fprintf(stderr,"Error: %s out of range (%d..%d) \n",param,min,max);
+    exit(VP8ENC_FAIL);
+  }
+}
+
+void parameter_check_positive(const char *param, int val, int min)
+{
+  if(val < 1)
+  {
+    fprintf(stderr,"Error: %s demands a positive value greater than %d \n",param,min);
+    exit(VP8ENC_FAIL);
+  }
+}
+
+int parse_options(int ac,char *av[])
+{
+  int c, long_index, tmp_input;
+  while (1) {
+      c = getopt_long_only(ac,av,"hf:?",long_opts,&long_index);
+
+      if (c == -1)
+          break;
+
+      switch(c) {
+      case 'f':
+          tmp_input = atoi(optarg);
+          parameter_check_positive("-f",tmp_input,1);
+          settings.frame_rate = tmp_input;
+          break;
+      case 1:
+          tmp_input = atoi(optarg);
+          parameter_check("--rcmode", tmp_input, 0, 2);
+          settings.rc_mode = default_rc_modes[tmp_input];
+          break;
+      case 2:
+          tmp_input = atoi(optarg);
+          parameter_check("--qb", tmp_input, 0, 255);
+          settings.quantization_parameter = tmp_input;
+          break;
+      case 3:
+          tmp_input = atoi(optarg);
+          parameter_check_positive("--intra_period",tmp_input,1);
+          settings.intra_period = tmp_input;
+          break;
+      case 4:
+          tmp_input = atoi(optarg);
+          parameter_check_positive("--fb",tmp_input,1);
+          settings.frame_bitrate = tmp_input;
+          break;
```

Comment:
Why no "case 5" ??

Reply:
i took the long_opts[] array from vp9enc - and for some reason there was no 5 - will fix it